src/unity/python/turicreate/visualization/show.py

```diff
@@ -96,46 +96,302 @@ def show(x, y, xlabel="X", ylabel="Y", title=None):
+    ------------
+    An object of Plot that is the columnwise summary
+
+    Examples
+    --------
+    Make a columnwise summary of an SFrame.
+
+    >>> x = turicreate.SArray([1,2,3,4,5])
+    >>> s = turicreate.SArray(['a','b','c','a','a'])
+    >>> sf_test = turicreate.SFrame([x,x,x,x,s,s,s,x,s,x,s,s,s,x,x])
+    >>> colsum = turicreate.visualization.columnwise_summary(sf_test)
+    """
+    if (type(sf) != tc.data_structures.sframe.SFrame):
+        raise ValueError("turicreate.visualization.columnwise_summary " + 
+            "supports SFrame")
     path_to_client = _get_client_app_path()
     plt_ref = tc.extensions.plot_columnwise_summary(path_to_client, sf)
     return Plot(plt_ref)
 
 def histogram(sa, xlabel="Values", ylabel="Count", title=None):
+    """
+    Plots a histogram of the sarray provided as input, and returns the 
+    resulting Plot object.
+    
+    The function supports numeric SArrays with dtypes int or float.
+
+    Parameters
+    ----------
+    sa : SArray
+      The data to get a histogram for. Must be numeric (int/float)
+    xlabel : str (optional)
+      The text label for the X axis. Defaults to "Values".
+    ylabel : str (optional)
+      The text label for the Y axis. Defaults to "Count".
+    title : str (optional)
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+    
+    Return Value
+    ------------
+    An object of Plot that is the histogram
+
+    Examples
+    --------
+    Make a histogram of an SArray.
+
+    >>> x = turicreate.SArray([1,2,3,4,5,1,1,1,1,2,2,3,2,3,1,1,1,4])
+    >>> hist = turicreate.visualization.histogram(x)
+    """
+    if (type(sa) != tc.data_structures.sarray.SArray or 
+        sa.dtype not in [int, float]):
+        raise ValueError("turicreate.visualization.histogram supports " + 
+            "SArrays of dtypes: int, float")
     path_to_client = _get_client_app_path()
     title = _get_title(title)
     plt_ref = tc.extensions.plot_histogram(path_to_client, sa, 
       xlabel, ylabel, title)
     return Plot(plt_ref)
 
 def item_frequency(sa, xlabel="Values", ylabel="Count", title=None):
+    """
+    Plots an item frequency of the sarray provided as input, and returns the 
+    resulting Plot object.
+    
+    The function supports SArrays with dtype str.
+
+    Parameters
+    ----------
+    sa : SArray
+      The data to get an item frequency for. Must have dtype str
+    xlabel : str (optional)
+      The text label for the X axis. Defaults to "Values".
+    ylabel : str (optional)
+      The text label for the Y axis. Defaults to "Count".
+    title : str (optional)
+      The title of the plot. Defaults to None. If the value is None, the title
+      will be "<xlabel> vs. <ylabel>". Otherwise, the string passed in as the
+      title will be used as the plot title.
+    
+    Return Value
```

Comment:
Same comment as for scatter plot (should be Returns, with name/type/description in special syntax).